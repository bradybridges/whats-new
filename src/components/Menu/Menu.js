import React from 'react';
import './Menu.scss'
import NavButton from '../NavButton/NavButton';

class Menu extends React.Component {
  
  returnTabs = () => {
    const tabs = ['local','health','science','technology','entertainment']
    return tabs.map((tab, i) => <NavButton 
      updateCurrentCategory={this.props.updateCurrentCategory} 
      key={i} name={tab}
    />)
  }

  render() {
    return (
      <nav>
        {this.returnTabs()}
      </nav>
    )
  }
}

export default Menu;