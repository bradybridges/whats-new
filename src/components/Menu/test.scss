$primaryColor : blue;

button {
  background-color: $primaryColor;
}