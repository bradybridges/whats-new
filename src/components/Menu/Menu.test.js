import React from 'react';
import ReactDOM from 'react-dom';
import Menu from './Menu';
import { shallow } from 'enzyme';

describe('Menu', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Menu updateCurrentCategory={jest.fn()} />)
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire function when clicked', () => {
    console.log(wrapper.debug())
  });

  describe('returnTabs', () => {
    it('should return all of the NavButtons', () => {
      expect(wrapper.instance().returnTabs().length).toEqual(5);
    });
  });  
})//EOF