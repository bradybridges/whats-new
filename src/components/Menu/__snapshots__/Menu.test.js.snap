// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<nav>
  <NavButton
    key="0"
    name="local"
    updateCurrentCategory={[MockFunction]}
  />
  <NavButton
    key="1"
    name="health"
    updateCurrentCategory={[MockFunction]}
  />
  <NavButton
    key="2"
    name="science"
    updateCurrentCategory={[MockFunction]}
  />
  <NavButton
    key="3"
    name="technology"
    updateCurrentCategory={[MockFunction]}
  />
  <NavButton
    key="4"
    name="entertainment"
    updateCurrentCategory={[MockFunction]}
  />
</nav>
`;
